Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Utilities/MaterialInstancing.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Microsoft.MixedReality.Toolkit.UI
+{
+    public class MaterialInstancing : MonoBehaviour
```

Comment:
Add comments why we need this. Because can't do materialpropertyblock and modification of material properties acts on shared material etc

Reply:
suggestion: maybe UIMaterialInstancing?

I just want to really highlight that people shouldn't be using this for Renderer components, only for Unity UI images etc

Reply:
I also added a requiremecomponent attribute that will prevent people from using it outside of this scope